### PR TITLE
Autostash for git svn rebase

### PR DESCRIPTION
**Pull request by renelink**
Hi,

I extended the git svn rebase command to support autostash.
It would be nice if this feature would be integrated in git soon.
Please review my changes.

Regards
René Link

### CODE REVIEW

**Comment by pranitbauva1997** [comment]
@renelink Git uses a different approach of accepting patches as described in Documentation/SubmittingPatches . Alternatively you can also use https://submitgit.herokuapp.com/ . This can send PR as a email to the mailing list in the form of a patch. Thanks!

**Comment by dscho** [comment]
Indeed. @renelink please send the patch to the mailing list for review. You can use [GitGitGadget](https://gitgitgadget.github.io), [submitGit](https://submitgit.herokuapp.com) or [send it manually](https://github.com/git-for-windows/git/blob/master/CONTRIBUTING.md#submit-your-patch).

**Comment by dscho** [comment]
@renelink gentle ping?

**Comment by dscho** [comment]
@renelink gentle ping?

**Comment by renelink** [comment]
@dscho I will take a look at it soon. Maybe my patch is not compatible with the actual master and I have to reintegrate it.

**Comment by dscho** [comment]
> Maybe my patch is not compatible with the actual master and I have to reintegrate it.

It would appear that this PR shows no merge conflicts. So I don't think you need to reintegrate (or, in Git speak, "rebase") it.

However, please do make sure that your commit message focuses on answering the question "why?" more than on "how?", and that it wraps at <= 76 columns per line. Also do make sure to add your [sign off](https://git-scm.com/docs/SubmittingPatches#sign-off).

Finally send the patch to the mailing list for review. You can use [GitGitGadget](https://gitgitgadget.github.io), [submitGit](https://submitgit.herokuapp.com) or [send it manually](https://github.com/git-for-windows/git/blob/master/CONTRIBUTING.md#submit-your-patch).

**Comment by dscho** [comment]
@renelink gentle ping?

**Comment by dscho** [comment]
@renelink another gently ping? Your commit message still should be changed to focus on answering the question "why?" more than on "how?", and to wrap at <= 76 columns per line. Also please do make sure to add your [sign off](https://git-scm.com/docs/SubmittingPatches#signoff).

**Comment by dscho** [comment]
@renelink maybe you can simply rebase to current master and force-push, just to get the ball rolling again?

**Comment by dscho** [comment]
@renelink ping?

**Comment by dscho** [comment]
Maybe you find some time over the holidays, @renelink?

File: git-svn.perl [hunk 1]
```diff
@@ -1247,7 +1247,8 @@ sub auto_create_empty_directories {
 }
 
 sub cmd_rebase {
-	command_noisy(qw/update-index --refresh/);
+    my $autostashCommitId;
```
Comment (peres57) [inline git-svn.perl]: 142215

Comment (peres57) [reply git-svn.perl]: 142215

File: git-svn.perl [hunk 2]
```diff
@@ -1247,7 +1247,8 @@ sub auto_create_empty_directories {
...
+	my $autostashEnabled = "0";
```
Comment (peres57) [inline git-svn.perl]: 142215

**Comment by dscho** [comment]
@renelink happy new year! Maybe this year you will find time to address the feedback in this PR?

**Comment by dscho** [comment]
@renelink gentle ping?

**Pull request closed without merging**